File: frontend/src/types/category.ts
export interface Category {
    id: number;
    name: string;
    slug: string;
    description: string;
    image_url?: string | null;
    parent_id?: number | null;

}

export interface CategoriesListResponse {
    ok: boolean;
    data: {
        page: number;
        limit: number;
        total: number;
        items: Category[];
    };
}

File: models/category.go
package models

import "gorm.io/gorm"

type Category struct {
	gorm.Model
	Name        string `gorm:"size:100;not null"`
	Slug        string `gorm:"size:100;uniqueIndex;not null"`
	Description string `gorm:"type:text"`

	ImageURL string `gorm:"size:500"`

	ParentID *uint
	Parent   *Category `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`

	Children []Category `gorm:"foreignKey:ParentID"`

	Products []Product
}
